libc/unistd_uids.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Testing getuid functions form unistd.h
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <stdlib.h>
+
+#include <unity_fixture.h>
+
+
+pid_t child[6], parent[3], pid;
+
+
+TEST_GROUP(unistd_uids);
+
+TEST_SETUP(unistd_uids)
+{
+	child = NULL;
+	pid = parent[0] = parent[1] = parent[2] = -1;
+}
+
+
+TEST_TEAR_DOWN(unistd_uids)
+{
+	free(child);
+}
+
+/*
+	Testing correct return values for:
+	getpid, getppid, getuid(), geteuid, 
+	getpgid, getpgrp, getsid, getgid, getegid
+
+	for self standing process
+*/
+TEST(unistd_uids, getuids_parent)
+{
+	TEST_ASSERT_GREATER_THAN_INT(0, getpid());
+	TEST_ASSERT_GREATER_THAN_INT(0, getppid());
+
+	pid = getpid();
+	TEST_ASSERT_GREATER_THAN_INT(0, getpgid(pid));
+	TEST_ASSERT_GREATER_THAN_INT(0, getpgrp());
+	TEST_ASSERT_GREATER_THAN_INT(0, getsid(pid));
+
+	/*
+		tests are run as root, but following functions are
+		declared but unimplemented in libphoenix.
+		Reference: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/280
+	*/
+	TEST_IGNORE();
+	TEST_ASSERT_GREATER_THAN_INT(0, getgid());
+	TEST_ASSERT_GREATER_THAN_INT(0, getegid());
+	TEST_ASSERT_EQUAL_INT(0, getuid());
+	TEST_ASSERT_EQUAL_INT(0, geteuid());
+}
+
+TEST(unistd_uids, setuids_parent)
+{
+	/*
+		setuid(), seteuid(), setgid(), setegid() are stub implemented so testing them
+		is pointless, yet marking this in testing process should occur.
+		Reference: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/280
+	*/
+	TEST_IGNORE();
+}
+
+TEST(unistd_uids, setpuids_setsid)
+{
+	int err, siderr = -1;
+
+	parent[0] = getpid();
+	parent[1] = getpgrp();
+	parent[2] = getsid(getpid());
+
+	child = malloc(sizeof(pid_t) * 6);
```

Comment:
```suggestion
```

Reply:
Yes. I am correcting that. Also `free(child)` is to be removed. Thanks!